custom-configs/src/main/java/com/sequenceiq/cloudbreak/service/CustomConfigsService.java

```diff
@@ -0,0 +1,225 @@
+package com.sequenceiq.cloudbreak.service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.cloudera.api.swagger.model.ApiClusterTemplateRoleConfigGroup;
+import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.service.ResourcePropertyProvider;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.auth.crn.CrnResourceDescriptor;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareCrnGenerator;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.domain.CustomConfigProperty;
+import com.sequenceiq.cloudbreak.domain.CustomConfigs;
+import com.sequenceiq.cloudbreak.exception.CustomConfigsExistsException;
+import com.sequenceiq.cloudbreak.repository.CustomConfigsRepository;
+
+@Service
+public class CustomConfigsService implements ResourcePropertyProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CustomConfigsService.class);
+
+    @Inject
+    private CustomConfigsRepository customConfigsRepository;
+
+    @Inject
+    private CustomConfigsValidator validator;
+
+    @Inject
+    private RegionAwareCrnGenerator regionAwareCrnGenerator;
+
+    public String createCRN(String accountId) {
+        return regionAwareCrnGenerator.generateCrnStringWithUuid(CrnResourceDescriptor.CUSTOM_CONFIGS, accountId);
+    }
+
+    public void decorateWithCrn(CustomConfigs customConfigs, String accountId) {
+        customConfigs.setResourceCrn(createCRN(accountId));
+    }
+
```

Comment:
can be merged with above method as initializeCrnForCustomConfig

Reply:
Merged into a single function